backend/dataall/cdkproxy/assets/lakeformationdefaultsettings/index.py

```diff
@@ -1,17 +1,39 @@
 import os
 import boto3
 from botocore.exceptions import ClientError
+import logging
+
+logger = logging.getLogger()
+logger.setLevel(os.environ.get("LOG_LEVEL", "INFO"))
+log = logging.getLogger(__name__)
+
+AWS_ACCOUNT = os.environ.get('AWS_ACCOUNT')
+AWS_REGION = os.environ.get('AWS_REGION')
+lf_client = boto3.client("lakeformation", region_name=os.environ.get("AWS_REGION"))
+iam_client = boto3.client('iam')
 
 
 def clean_props(**props):
     data = {k: props[k] for k in props.keys() if k != 'ServiceToken'}
     return data
 
 
+def validate_principals(principals):
+    validated_principals = []
+    for principal in principals:
+        if ":role/" in principal:
+            log.info(f'Principal {principal} is an IAM role, validating....')
+            try:
+                iam_client.get_role(RoleName=principal.split("/")[-1])
+                log.info(f'Adding principal {principal} to validated principals')
+                validated_principals.append(principal)
+            except Exception as e:
+                log.exception(f'Failed to get role {principal} due to: {e}')
+                raise Exception(f'Failed to get role {principal} due to: {e}')
```

Comment:
Wasn't the intent here to return the list of valid principals, ignoring the errors? Otherwise `on_create`/`on_delete` will fail.

Reply:
Yeeep, the exception should only be logged. Thank you